code-generators/npm-schema-only-package/src/index.mjs

```diff
@@ -0,0 +1,60 @@
+import * as fs from "fs";
+import {mkdir, readdir, writeFile, copyFile} from "fs/promises";
+import * as path from "path";
+
+const args = process.argv.slice(2);
+if (args.length < 1) {
+    throw new Error("Missing path to JSON schema")
+}
+
+async function generate(schemaDir, genDir) {
+    const files = (await readdir(schemaDir))
+        .filter((f) => f.endsWith(".json"));
+
+    console.log(`${files.length} schema files`, files);
+
+    let indexFile = path.join(genDir, 'index.d.ts');
+    let index = "";
+    let exportList = [];
+    for (const file of files) {
+        const schemaFile = path.join(schemaDir, file);
+        console.log(`Copying schema ${schemaFile}`);
+
+        const schemaName = path.basename(schemaFile);
+        await copyFile(schemaFile, path.join(genDir, 'schemas', schemaName));
+        const schemaExportName = path.basename(schemaName, '.json');
+        exportList.push(schemaExportName);
+        index += `import ${schemaExportName} from'./schemas/${schemaName}';\n`
+    }
+
+    index += `\nmodule.exports = [${exportList.join(', ')}];\n`;
+    await writeFile(indexFile, index);
+
+    console.log(`Wrote schemas files to: ${genDir}`);
+}
+
+const schemaDir = args[0];
+console.log(`Generating from schema dir: ${schemaDir}`);
+
+const genDir = args.length >= 2 ? args[1] : path.join(process.cwd(), "gen");
+if (!fs.existsSync(genDir)) {
+    await mkdir(genDir);
+    await mkdir(path.join(genDir, 'schemas'));
+}
+
+generate(schemaDir, genDir)
+    .catch((cause) => console.error(cause));
+
+await writeFile(path.join(genDir, 'package.json'), JSON.stringify({
```

Comment:
See comment in https://github.com/govuk-one-login/data-vocab/pull/111#discussion_r1618371937 which may be a better alternative.

Reply:
Weirdly, I did it this way to avoid have the external scripts and template, thought it was neater to wrap up into the generation script 🤔